tests/unit/amo/api/test_users.js

```diff
@@ -252,4 +253,37 @@ describe(__filename, () => {
       mockApi.verify();
     });
   });
+
+  describe('unsubscribeNotification', () => {
+    it('unsubscribes a user from a notification', async () => {
+      const { state } = dispatchClientMetadata();
+      const hash = 'some-hash';
+      const token = 'some-token';
+      const notification = 'new_review';
+      const params = {
+        api: state.api,
+        hash,
+        notification,
+        token,
+      };
+
+      mockApi
+        .expects('callApi')
+        .withArgs({
+          apiState: params.api,
+          auth: false,
```

Comment:
I know this doesn't really have to do with this specific patch, but it seems odd to me that the `unsubscribe` API does not require authentication. Doesn't that mean that anyone can unsubscribe anyone else from a notification? I guess you'd need the correct `hash` and `token`, so is that what makes it safe?

Reply:
Yep, exactly, there is no other authentication than the hash/token couple.

It's important that these links work in all situations and I hate when I have to find 10 years old credentials to unsubscribe from a mailing-list.